org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/typeHierarchy/TypeHierarchyView.java

```diff
@@ -74,21 +385,99 @@ private void updateHierarchyModeItem(ToolItem hierchyModeItem, boolean showSuper
 				hierchyModeItem.setToolTipText(showSuperTypes ? Messages.typeHierarchy_show_subtypes : Messages.typeHierarchy_show_supertypes);
 			}
 		};
-		treeViewer = filteredTree.getViewer();
-		treeViewer.setContentProvider(contentProvider);
+	}
 
-		treeViewer.setLabelProvider(new TypeHierarchyItemLabelProvider());
+	private SymbolsContainer getSymbolsContainer(IFile file) {
+		SymbolsContainer symbolsContainer = cachedSymbols.get(file.getLocationURI());
```

Comment:
instead of using  `cachedSymbols.get` and  `cachedSymbols.put`, could we not use  `cachedSymbols.computeIfAbsent`?